0000-stream/0000-stream.md

```diff
@@ -0,0 +1,174 @@
+
+If subsequent versions support additional encryption algorithms, those details should be exchanged between the sender and receiver when they establish the shared secret. If a receiver attempts to decrypt an incoming packet but is unable to (perhaps because the sender is using an unsupported cipher), they SHOULD reject the incoming transfer with an `F06: Unexpected Payment` error.
+
+### Conditions and Fulfillments
+
+The Client and Server generate the condition and fulfillment using their shared secret. The fulfillment is computed from the shared secret and the encrypted STREAM packet. The condition is the hash of the fulfillment.
+
+As in PSKv1 and PSKv2, this method for generating the condition and fulfillment enables many payments to be sent without end-to-end communication for each one.
+
+### Quoting
+
+STREAM includes the prepare amount in each packet to enable the Client and Server to determine and handle the path exchange rate. STREAM packets sent with ILP Prepare packets include the minimum destination amount the other party should accept, which the sender may set using the computed exchange rate. STREAM packets included in ILP Fulfill and Reject packets include the amount that arrived in the ILP Prepare packet the receiver got, which enables the sender to determine the exchange rate by dividing the destination amount by the source amount they sent.
+
+### Differences from PSK
+
+Unlike PSK and PSK2, STREAM handles:
+- Streaming and chunked payments
+- Segmenting application data to send it using multiple ILP packets
+
+### Differences from QUIC
+
+Unlike QUIC, STREAM:
+- Has only one packet type and the shared secret identifies the Connection rather than a separate Connection ID.
+- Does not include a cryptographic handshake, because STREAM assumes a symmetric secret is communicated out of band.
+- Does not support unidirectional frames. The QUIC community had significant debate over whether to include unidirectional streams, bidirectional streams, or both. They settled on both primarily to support the HTTP request/response pattern as well as HTTP/2 Server Push. Unidirectional streams were left out of STREAM because they add complexity and are a premature optimization for this protocol now.
+- Does not have ACK frames, because ILP Prepare packets must be acknowledged with either a Fulfill or Reject packet. If the response includes an (authenticated) STREAM packet, the sender can treat that as an acknowledgement of the control and data frames from the Prepare packet they sent.
+- Does not have Ping, Path Challenge, or Path Challenge Response frames, because a STREAM packet with no frames can be used instead. As long as the Client and Server increment the packet sequence for each packet they send, a valid Fulfill or Reject packet from the other party that includes the correct sequence in the encrypted data serves as the path challenge and response.
```

Comment:
What does the packet sequence do exactly, and are there any ways it could get out of sync?